pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1040,29 +1006,91 @@ func (pl *dynamicResources) Unreserve(ctx context.Context, cs *framework.CycleSt
 	}
 
 	logger := klog.FromContext(ctx)
-	for index, claim := range state.claims {
+
+	// Was publishing delayed? If yes, do it now.
+	if state.podSchedulingState.isDirty() {
+		if err := state.podSchedulingState.publish(ctx, pod, pl.clientset); err != nil {
+			logger.Error(err, "publish PodSchedulingContext")
+		}
+	}
```

Comment:
Should we publish here?
The function `podSchedulingState.isDirty` does not check delta and simply return true if some fields are populated.
So publish will be called.
1.  Even though it was not called due to the failure of Reserve plugins.
1.  Even though it was called in the previous step: PreBind.

In the first scenario, we already know a node is not suitable, could we get rid of the node from selected node or potential nodes? since It would trigger driver to call `UnsutableNodes` and `allocateClaims`. Maybe calling out  `UnsutableNodes` is necessary but we may no need to push driver call `allocateClaims`since we already know we don't need to do that.

Reply:
> So publish will be called.

Yes, but only if there really are pending changes. The fields will be unset if nothing needs to be changed. The code that updates `podSchedulingState` is careful to only set fields which really need to be published.

> Even though it was called in the previous step: PreBind.

`publish` clears fields when it is done, so publish will not get called again.